kernel/aio.c

```diff
@@ -0,0 +1,448 @@
+#include "debug.h"
+#include "kernel/calls.h"
+#include "kernel/task.h"
+#include "kernel/aio.h"
+#include "kernel/fs.h"
+#include "kernel/time.h"
+#include "fs/aio.h"
+#include "fs/fd.h"
+
+// Guest memory offsets for the IOCB structure.
+// Calculated by a test program compiled and ran in iSH itself.
+const size_t AIO_IOCB_DATA = 0;
```

Comment:
Why not define a struct, instead of reading and writing individual offsets? Then these could become static_assert lines to verify that the struct layout is correct.